prime-router/src/main/kotlin/transport/SftpTransport.kt

```diff
@@ -43,7 +43,16 @@ class SftpTransport : ITransport, Logging {
             if (header.content == null)
                 error("No content to sftp for report ${header.reportFile.reportId}")
             val receiver = header.receiver ?: error("No receiver defined for report ${header.reportFile.reportId}")
-            val credential = lookupCredentials(receiver.fullName)
+
+            // if the transport definition has defined default
+            // credentials use them, otherwise go with the
+            // standard way by using the receiver full name
+            val credential = if (sftpTransportType.credentials != null) {
+                getCredentials(sftpTransportType.credentials)
+            } else {
+                lookupCredentials(receiver.fullName)
+            }
```

Comment:
Instead of having discrete methods, I would opt to reuse the same method. The logic on lines 97-100 are to sanitize the string for unsafe characters in Azure. Therefore, if the method signature is changed to:

```
fun lookupCredentials(credentialName: String): SftpCredential
```

then the `getCredentials` method could be removed and the following lines simplified with an elvis:

```suggestion
            val credential = lookupCredentials(sftpTransportType?.credentials ?: receiver.fullName)
```

Reply:
fixed